aclmapping/evm/mappings.go

```diff
@@ -0,0 +1,95 @@
+package evm
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkacltypes "github.com/cosmos/cosmos-sdk/types/accesscontrol"
+	aclkeeper "github.com/cosmos/cosmos-sdk/x/accesscontrol/keeper"
+	acltypes "github.com/cosmos/cosmos-sdk/x/accesscontrol/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	evmkeeper "github.com/sei-protocol/sei-chain/x/evm/keeper"
+	evmtypes "github.com/sei-protocol/sei-chain/x/evm/types"
+)
+
+var ErrInvalidMessageType = fmt.Errorf("invalid message received for EVM Module")
+
+func GetEVMDependencyGenerators(evmKeeper evmkeeper.Keeper) aclkeeper.DependencyGeneratorMap {
+	dependencyGeneratorMap := make(aclkeeper.DependencyGeneratorMap)
+	EVMTransactionMsgKey := acltypes.GenerateMessageKey(&evmtypes.MsgEVMTransaction{})
+	dependencyGeneratorMap[EVMTransactionMsgKey] = func(k aclkeeper.Keeper, ctx sdk.Context, msg sdk.Msg) ([]sdkacltypes.AccessOperation, error) {
+		return TransactionDependencyGenerator(k, evmKeeper, ctx, msg)
+	}
+
+	return dependencyGeneratorMap
+}
+
+func TransactionDependencyGenerator(_ aclkeeper.Keeper, evmKeeper evmkeeper.Keeper, ctx sdk.Context, msg sdk.Msg) ([]sdkacltypes.AccessOperation, error) {
+	evmMsg, ok := msg.(*evmtypes.MsgEVMTransaction)
+	if !ok {
+		return []sdkacltypes.AccessOperation{}, ErrInvalidMessageType
+	}
+
+	tx, _ := evmMsg.AsTransaction()
+	// Only specifying accesses to `to` address since `from` has to be derived via signature,
+	// which happens in the ante handler (i.e. after this generator is called) and is quite heavy
+	// so we don't want to repeat it.
+	toAddress := tx.To()
+	seiAddress, associated := evmKeeper.GetSeiAddress(ctx, *toAddress)
+	var idTempl string
+	var resourceType sdkacltypes.ResourceType
+	if associated {
+		idTempl = hex.EncodeToString(banktypes.CreateAccountBalancesPrefix(seiAddress))
+		resourceType = sdkacltypes.ResourceType_KV_BANK_BALANCES
+	} else {
+		idTempl = hex.EncodeToString(evmtypes.BalanceKey(*toAddress))
+		resourceType = sdkacltypes.ResourceType_KV_EVM_BALANCE
```

Comment:
@codchen you need to update the github workflow to not run forward compatibility on `evm` branch (similar to how we did for occ), that way you can continue to have a working/accurate CI